### PR TITLE
Cleaner generated code

### PR DESCRIPTION
**Pull request by sharwell**
This change updates the compiler code generator to produce code that the formatter would not change.

The **IdeBenchmarks** project uses these large generated files as a formatting stress test. Unfortunately, the benchmark was not measuring the critical fast path that we thought it was because the generated code needed changes applied. Following this change, the formatter benchmark defaults to execution on code that does not require any changes as a result of the formatting operation.

### CODE REVIEW

File: src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/RedNodes/RedNodeWriter.vb [hunk 1]
```diff
@@ -598,78 +599,78 @@ Friend Class RedNodeWriter
...
             _writer.WriteLine("        End Function")
             _writer.WriteLine()
 
         ElseIf isOverride Then
 
-            _writer.WriteLine("        Public {0} ReadOnly Property {1} As {2}", If(isOverride, "Shadows", ""), ChildPropertyName(child), ChildPropertyTypeRef(nodeStructure, child))
+            _writer.WriteLine("        Public {0}ReadOnly Property {1} As {2}", If(isOverride, "Shadows ", ""), ChildPropertyName(child), ChildPropertyTypeRef(nodeStructure, child))
             _writer.WriteLine("            Get")
-            Me.GenerateNodeChildPropertyRedAccessLogic(nodeStructure, child, childIndex, isOverride)
+            Me.GenerateNodeChildPropertyRedAccessLogic(nodeStructure, child, childIndex, isOverride, extraIndent:="    ")
```
Comment (CyrusNajmabadi) [inline src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/RedNodes/RedNodeWriter.vb:611]: a little confused by this.  is this not just indenting/dedenting around the call to GenerateNodeChildPropertyRedAccessLogic?

Comment (CyrusNajmabadi) [reply src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/RedNodes/RedNodeWriter.vb:611]: oh. n/m .  this doesn't use any special helper to write this stuff out.

**Comment by RikkiGibson** [comment]
is there any validation in place that will start failing the build if the generator or formatter change such that the formatter *would* format it?

**Comment by sharwell** [comment]
@RikkiGibson there is not, but I'm not sure how that would be done or to what degree it would matter in the end.

**Comment by RikkiGibson** [comment]
Run the formatter on the files and fail if there are changes?

I’m just concerned that if maintaining this invariant is important for the benchmarks to be as meaningful as possible, then we should assume that the generator and formatter will “desync” again without some automated validation.

It might make sense to make copies of a few huge but well formatted C# files to use as stress tests which either don’t change or do have this validation performed on them.

File: src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/AbstractFileWriter.cs [hunk 2]
```diff
@@ -75,6 +75,12 @@ protected void WriteLine(string msg)
             _needIndent = true; //need an indent after each line break
         }
 
+        protected void WriteLineWithoutIndent(string msg)
```
Comment (333fred) [inline src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/AbstractFileWriter.cs:78]: This naming is just confusing. The name is `WriteLineWithoutIndent`, and yet is specifically sets `_needIndent`.

File: src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs [hunk 3]
```diff
@@ -1077,7 +1077,7 @@ private void WriteVisitor()
                     Indent();
                     foreach (var node in _tree.Types.OfType<Node>())
                     {
-                        WriteLine("Case BoundKind.{0}: ", FixKeyword(StripBound(node.Name)));
+                        WriteLine("Case BoundKind.{0} :", FixKeyword(StripBound(node.Name)));
```
[the remark this answers is not in the document]
Comment (sharwell) [reply src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs:1080]: You may be thinking, as I was originally, that this `:` is part of the case label. You would be incorrect. 😂

File: src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/AbstractFileWriter.cs (hunk 2 above)
Comment (sharwell) [reply src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/AbstractFileWriter.cs:78]: The current line is written without indent. The next line is not. Open to alternate names if you have a suggestion.

File: src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs [hunk 4]
```diff
@@ -1283,7 +1283,7 @@ private void WriteTreeDumperNodeProducer()
                             {
                                 Field field = allFields[i];
                                 if (IsDerivedType("BoundNode", field.Type))
-                                    Write("New TreeDumperNode(\"{0}\", Nothing, new TreeDumperNode() {{ Visit(node.{1}, Nothing) }})", ToCamelCase(field.Name), field.Name);
+                                    Write("New TreeDumperNode(\"{0}\", Nothing, new TreeDumperNode() {{Visit(node.{1}, Nothing)}})", ToCamelCase(field.Name), field.Name);
```
[the remark this answers is not in the document]
Comment (sharwell) [reply src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs:1286]: The VB formatter is not configurable. This is just what it does for collection initialization.

**Comment by sharwell** [comment]
> Seems like the job for part of the correctness leg.

I've requested a new flag for **dotnet-format** that allows it to run the formatter on generated code. This will allow us to validate the format as part of the correctness leg in a follow-up.

**Comment by jcouv** [comment]
>                 Case BoundKind.TypeArguments :

> You may be thinking, as I was originally, that this : is part of the case label. 

@sharwell I don't understand. Is this the formatting we're expecting for VB?

---
Refers to: src/Compilers/VisualBasic/Portable/Generated/BoundNodes.xml.Generated.vb:9285 in 4068586. [](commit_id = 406858600d4bfa503b1e765b048f56f5f05cc0fa, deletion_comment = False)

**Comment by jcouv** [comment]
>             Me.DefaultVisit(node) : Return

Do we need the ` : Return` at all?

---
Refers to: src/Compilers/VisualBasic/Portable/Generated/Syntax.xml.Main.Generated.vb:750 in 4068586. [](commit_id = 406858600d4bfa503b1e765b048f56f5f05cc0fa, deletion_comment = False)

**Comment by jcouv** [comment]
>                 Case SyntaxKind.IfKeyword :

I'm surprised here too

---
Refers to: src/Compilers/VisualBasic/Portable/Generated/Syntax.xml.Main.Generated.vb:5730 in 4068586. [](commit_id = 406858600d4bfa503b1e765b048f56f5f05cc0fa, deletion_comment = False)

**Comment by sharwell** [comment]
> I don't understand. Is this the formatting we're expecting for VB?

Yes. The `:` is not part of the case label so there is a space between the two.

**Comment by jcouv** [comment]
@sharwell I still don't follow :-/

Here's a simple [example](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBsQDdhJiaMkAmIA1AD4CSYADgPYIxQAEAygJ6xxgCwAUAAoBXYCgCWAY0YBhFNCaTejRY0HDxLIYwCyAChqMAgkwDywAFZwxMAJQKld5nBQWYjPbbuLJ0OIwCMIdw87ACU4GAEEADtAoMUYpQBRSIIWR2d4xkCklOYhXmypGSgoIA=) with `Case 1:`.

I think that by "not part of the case label" you mean that it's trivia following the case label.

image

If that's what you mean, I still don't see how that informs our recommended formatting. I don't see how the syntax structure restricts us one way or the other.

**Comment by CyrusNajmabadi** [comment]
> If that's what you mean, I still don't see how that informs our recommended formatting. I don't see how the syntax structure restricts us one way or the other.

It's more that historically we have formatted colon-trivia this way.  So that's that.  The syntax point was to mention that it's not like c# which has `case foo:`, in vb it's jsut `case foo` and the `:` here really is something different, and this different thing is formatted this way by the VB formatter (likely because it's always done that with trivia).

**Comment by sharwell** [comment]
Here's what a validation failure looks like in CI:

```
  Formatting code files in workspace 'F:\workspace\_work\1\s'.
  s\src\Compilers\CSharp\Portable\Generated\BoundNodes.xml.Generated.cs(7849,56): Fix whitespace formatting.
  Formatted code file 'F:\workspace\_work\1\s\src\Compilers\CSharp\Portable\Generated\BoundNodes.xml.Generated.cs'.
  Format complete in 22844ms.
Command failed to execute with exit code 2: dotnet format . --include-generated --include src/Compilers/CSharp/Portable/Generated/ src/Compilers/VisualBasic/Portable/Generated/ src/ExpressionEvaluator/VisualBasic/Source/ResultProvider/Generated/ --check -f
```

File: NuGet.config [hunk 5]
```diff
@@ -12,6 +12,7 @@
     <!-- Microsoft.CodeAnalysis.Test.Resource.Proprietary -->
     <add key="roslyn-tools" value="https://dotnet.myget.org/F/roslyn-tools/api/v3/index.json" />
     <add key="roslyn-analyzers" value="https://dotnet.myget.org/F/roslyn-analyzers/api/v3/index.json" />
+    <add key="dotnet-format" value="https://dotnet.myget.org/F/format/api/v3/index.json" />
```
Comment (JoeRobich) [inline NuGet.config:15]: This is no longer necessary as the referenced version of dotnet-format is available from nuget.

Comment (sharwell) [reply NuGet.config]: Removed this change.

**Comment by sharwell** [comment]
@RikkiGibson @333fred can one of you provide a second review now that I've enabled CI validation to avoid future regressions?

**Comment by sharwell** [comment]
Ping @RikkiGibson @333fred

File: src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (sharwell) [reply src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs:1087]: 📝 The `:` is a statement separator in VB, which would be used if we wanted to combine the lines but as a trailing separator it doesn't have any impact. I avoided making any real changes in this PR, opting instead to format whitespace and enable validation. Users who want to make improvements to the generated code can now do so easily.

[the remark this answers is not in the document]
Comment (333fred) [reply src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs:1087]: We should either use `:` for its intended purpose (to allow the `Return` statement on the same line) or just remove it. It's fine if you don't want to address it in this PR, I can send a follow up.

**Comment by sharwell** [comment]
@333fred I'll prepare a follow-up to remove the unnecessary `:` characters, and submit it for review right after I merge this one.

File: src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (sharwell) [reply src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs:1087]: I will submit a follow-up PR to eliminate this unnecessary colon.

**Pull request merged**